src/accountsdb/snapshots.zig

```diff
@@ -326,22 +327,34 @@ pub const BankFields = struct {
 
 /// Analogous to [SerializableAccountStorageEntry](https://github.com/anza-xyz/agave/blob/cadba689cb44db93e9c625770cafd2fc0ae89e33/runtime/src/serde_snapshot/storage.rs#L11)
 pub const AccountFileInfo = struct {
-    // note: serialized id is a usize but in code its FileId (u32)
-    id: usize,
+    // note: serialized id is a usize but in code it's FileId (u32)
+    id: FileId,
     length: usize, // amount of bytes used
 
+    pub const @"!bincode-config:id": bincode.FieldConfig(FileId) = .{
+        .serializer = idSerializer,
+        .deserializer = idDeserializer,
+    };
+
+    fn idSerializer(writer: anytype, data: anytype, params: bincode.Params) anyerror!void {
+        try bincode.write(writer, @as(usize, data.toInt()), params);
+    }
+
+    fn idDeserializer(_: std.mem.Allocator, reader: anytype, params: bincode.Params) anyerror!FileId {
+        var fba = comptime std.heap.FixedBufferAllocator.init(&.{});
```

Comment:
why not just use the allocator parameter thats passed in to this fcn?

Reply:
the point is that it should fail if it tries to allocate, because it should never allocate - we're deserializing an integer.

Reply:
There should maybe be a dedicated function for this.

Reply:
makes sense - yeah dedicated function `readWithoutAlloc` might be a good bincode function addition which just calls `read` with the fb allocator you have here

Reply:
I decided to go with a dedicated function for reading ints - the `readWithoutAlloc` idea is interesting, but it seems hard to verify such a thing in the general case.